sql-plugin/src/main/scala/com/nvidia/spark/rapids/datetimeRebaseUtils.scala

```diff
@@ -23,9 +23,88 @@ import com.nvidia.spark.rapids.shims.SparkShimImpl
 
-object RebaseHelper {
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy
+ */
+sealed abstract class DateTimeRebaseMode(val value: String) extends Serializable
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.EXCEPTION
+ */
+case object DateTimeRebaseException extends DateTimeRebaseMode("EXCEPTION")
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.LEGACY
+ */
+case object DateTimeRebaseLegacy extends DateTimeRebaseMode("LEGACY")
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.CORRECTED
+ */
+case object DateTimeRebaseCorrected extends DateTimeRebaseMode("CORRECTED")
+
+object DateTimeRebaseUtils {
+  def invalidRebaseModeMessage(name: String): String =
+    s"Invalid datetime rebase mode: $name (must be either 'EXCEPTION', 'LEGACY', or 'CORRECTED')"
+
+  def rebaseModeFromName(name: String): DateTimeRebaseMode = name match {
+    case DateTimeRebaseException.value => DateTimeRebaseException
+    case DateTimeRebaseLegacy.value => DateTimeRebaseLegacy
+    case DateTimeRebaseCorrected.value => DateTimeRebaseCorrected
+    case _ => throw new IllegalArgumentException(invalidRebaseModeMessage(name))
+  }
+
+  // Copied from Spark
+  private val SPARK_VERSION_METADATA_KEY = "org.apache.spark.version"
+  private val SPARK_LEGACY_DATETIME_METADATA_KEY = "org.apache.spark.legacyDateTime"
+  private val SPARK_LEGACY_INT96_METADATA_KEY = "org.apache.spark.legacyINT96"
+  private val SPARK_TIMEZONE_METADATA_KEY = "org.apache.spark.timeZone"
+
+  private def rebaseModeFromFileMeta(lookupFileMeta: String => String,
+                                     modeByConfig: String,
+                                     minVersion: String,
+                                     metadataKey: String): DateTimeRebaseMode = {
+
+    // If there is no version, we return the mode specified by the config.
+    val mode = Option(lookupFileMeta(SPARK_VERSION_METADATA_KEY)).map { version =>
+      // Files written by Spark 2.4 and earlier follow the legacy hybrid calendar and we need to
+      // rebase the datetime values.
+      // Files written by `minVersion` and latter may also need the rebase if they were written
+      // with the "LEGACY" rebase mode.
+      if (version < minVersion || lookupFileMeta(metadataKey) != null) {
+        DateTimeRebaseLegacy
+      } else {
+        DateTimeRebaseCorrected
+      }
+    }.getOrElse(rebaseModeFromName(modeByConfig))
+
+    // Check the timezone of the file if the mode is LEGACY.
+    if (mode == DateTimeRebaseLegacy) {
+      val fileTimeZone = lookupFileMeta(SPARK_TIMEZONE_METADATA_KEY)
+      if (fileTimeZone != null && fileTimeZone != "UTC") {
+        throw new UnsupportedOperationException(
+          "LEGACY datetime rebase mode is only supported for files written in UTC timezone. " +
+            s"Actual file timezone: $fileTimeZone")
+      }
+    }
+
+    mode
+  }
+
+  def datetimeRebaseMode(lookupFileMeta: String => String,
+                         modeByConfig: String): DateTimeRebaseMode = {
```

Comment:
Style guide does not align parameters regardless of indentation amount.  Continuation indent is 4 spaces.  If you want all the parameters aligned, write it this way:
```suggestion
  def datetimeRebaseMode(
      lookupFileMeta: String => String,
      modeByConfig: String): DateTimeRebaseMode = {
```

Reply:
My IntelliJ insists on aligning that way. Will try to fix it.